modules/tinymce/src/plugins/directionality/test/ts/browser/DirectionalitySanityTest.ts

```diff
@@ -9,42 +8,175 @@ describe('browser.tinymce.plugins.directionality.DirectionalitySanityTest', () =
     TinyUiActions.clickOnToolbar(editor, 'button[title="Left to right"]');
-    TinyAssertions.assertContentStructure(editor, ApproxStructure.build((s, str) => {
-      return s.element('body', {
-        children: [
-          s.element('p', {
-            attrs: {
-              dir: str.is('ltr')
-            }
-          })
-        ]
-      });
-    }));
+    TinyAssertions.assertContent(editor, '<p>a</p>'); // as the default dir is ltr it just removes the dir attr
+  });
+
+  it('TINY-4589: should set two paragraphs to rtl and ltl', () => {
+    const editor = hook.editor();
+    editor.setContent('<p>foo</p><p>bar</p>');
+    TinySelections.setSelection(editor, [ 0 ], 0, [ 1 ], 1);
+    TinyUiActions.clickOnToolbar(editor, 'button[title="Right to left"]');
+    TinyAssertions.assertContent(editor, '<p dir="rtl">foo</p><p dir="rtl">bar</p>');
+
+    TinySelections.setSelection(editor, [ 0 ], 0, [ 1 ], 1);
+    TinyUiActions.clickOnToolbar(editor, 'button[title="Left to right"]');
+    TinyAssertions.assertContent(editor, '<p>foo</p><p>bar</p>');
+  });
+
+  it('TINY-4589: should set parent dir when element is a list item', () => {
+    const editor = hook.editor();
+    editor.setContent('<ul><li>foo</li><li>bar</li></ul>');
+    TinySelections.setSelection(editor, [ 0, 0 ], 0, [ 0, 0 ], 1);
+    TinyUiActions.clickOnToolbar(editor, 'button[title="Right to left"]');
+    TinyAssertions.assertContent(editor, '<ul dir="rtl"><li>foo</li><li>bar</li></ul>');
+
+    TinyUiActions.clickOnToolbar(editor, 'button[title="Left to right"]');
+    TinyAssertions.assertContent(editor, '<ul><li>foo</li><li>bar</li></ul>');
+  });
+
+  it('TINY-4589: should remove dir from selected list item and children', () => {
+    const editor = hook.editor();
+    editor.setContent(
+      '<ul>' +
+        '<li dir="ltr">foo' +
+          '<ul>' +
+            '<li dir="ltr">a</li>' +
+            '<li dir="rtl">b</li>' +
+            '<li>c</li>' +
+          '</ul>' +
+        '</li>' +
+        '<li dir="xyz">bar</li>' +
+      '</ul>'
+    );
+    TinySelections.setSelection(editor, [ 0, 0 ], 0, [ 0, 0 ], 1);
+    TinyUiActions.clickOnToolbar(editor, 'button[title="Right to left"]');
+    TinyAssertions.assertContent(editor,
+      '<ul dir="rtl">' +
+        '<li>foo' +
+          '<ul>' +
+            '<li dir="ltr">a</li>' +
+            '<li dir="rtl">b</li>' +
+            '<li>c</li>' +
+          '</ul>' +
+        '</li>' +
+        '<li>bar</li>' +
+      '</ul>'
+    );
+  });
+
+  it('TINY-4589: should consider list item dir', () => {
+    const editor = hook.editor();
+    editor.setContent(
+      '<ul dir="rtl">' +
+        '<li dir="ltr">ini' +
+          '<ul>' +
+            '<li>foo</li>' +
+            '<li>bar</li>' +
+          '</ul>' +
+        '</li>' +
+      '</ul>'
+    );
+    TinySelections.setSelection(editor, [ 0, 0, 1, 0 ], 0, [ 0, 0, 1, 0 ], 1); // foo
+    TinyUiActions.clickOnToolbar(editor, 'button[title="Left to right"]');
```

Comment:
This test isn't really doing anything. Should this have been placed after the `rtl` change so we can check it moves back to `ltr`?

Reply:
The idea of this one is to assert that clicking on LTR when foo is selected won't change the ul direction as its parent has dir="ltr" already.

Reply:
That should be a separate case then, as that's testing something different to what's in the title (it's testing that applying the same dir makes no changes).

Reply:
Broke into two test cases: `should consider list item dir` and `applying the same dir makes no changes`.